comit/src/rsa.rs

```diff
@@ -0,0 +1,115 @@
+//! Recommended Safe Actions
+//!
+//! This module provides logic for recommending when it is safe/unsafe to take a
+//! particular action for the COMIT protocol.
+//!
+//! 'Safe' is defined as:
+//! 1 - You will not loose money.
+//! 2 - The counterparty is likely to continue with the swap because it is safe
+//!     for them to do so.
+
+use crate::Timestamp;
+use time::Duration;
+
+/// SwapTime enables recommendations on whether it is safe to take an action.
+#[derive(Debug, Clone, Copy)]
+pub struct SwapTime {
+    alpha_ledger: Ledger,
+    beta_ledger: Ledger,
+    alpha_tta: Duration,
+    beta_tta: Duration,
+    alpha_expiry: Timestamp,
+    beta_expry: Timestamp,
+}
+
+impl SwapTime {
+    /// Construct a new SwapTime from the given expiries.
+    pub fn new(
+        _alpha_ledger: Ledger,
+        _beta_ledger: Ledger,
+        _alpha_expiry: Timestamp,
+        _beta_expry: Timestamp,
+    ) -> Self {
+        unimplemented!()
+    }
+
+    /// Calculates a pair of valid expiries using the given parameters.
+    /// The created SwapTime will return true from calls to
+    /// is_time_for_swap_to_complete() for the duration of `good_for`.
+    pub fn with_calculated_expiries(
```

Comment:
From the name I assume this should take `self` by value.

Reply:
Good point, thanks.